src/core/blockchain/near/contracts/staking.ts

```diff
@@ -0,0 +1,129 @@
+import { Client } from '../client';
+import { IPosTransaction, IBlockchainTransaction, TransactionType } from '../../types';
+import { IValidator } from '../../types/stats';
+import { PosBasicActionType } from '../../types/token';
+import { buildBaseTransaction } from './base-contract';
+
+export class Staking {
+    constructor(private client: Client) {}
+
+    public async deposit(
+        tx: IPosTransaction,
+        validator: IValidator
+    ): Promise<IBlockchainTransaction> {
+        const transaction = await buildBaseTransaction(tx);
+
+        const contractAddress = validator.id;
+
+        transaction.amount = tx.amount;
```

Comment:
if the amount is not changed from what buildBaseTransaction sets don’t overwrite it